### ISSUE TITLE
System halted when calling `model.fit`

### ISSUE THREAD

**Issue by geekan** [fastai/fastai#751]
I tried lesson1 and lesson4-imdb jupyter notebook, however, **whenever I tried to train a model(calling `fit` method), the system halted and then rebooted.**

I tried to debug by myself, checked all system logs, and searched any suspicious log via everything, but none of them seems to log the error details.

**I notice that anaconda install cudnn-7.1.4-cuda9.0_0 and perhaps it's conflicting with current cuda?**

The error cell in `lesson1`
```
arch=resnet34
data = ImageClassifierData.from_paths(PATH, tfms=tfms_from_model(arch, sz))
learn = ConvLearner.pretrained(arch, data, precompute=True)
learn.fit(0.01, 2)
```

The error cell in `lesson4-imdb`
```
learner.fit(3e-3, 4, wds=1e-6, cycle_len=1, cycle_mult=2)
```

Running the error cells cause system halt and reboot.

```
os: windows 10
jupyter lab: 0.32.1
notebook: 5.5.0
cuda: 9.0.176

cudnn1(installed before): cudnn-9.0-windows10-x64-v7.2.1.38
cudnn2(installed by anaconda): cudnn-7.1.4-cuda9.0_0
- E:\Anaconda3\pkgs\cudnn-7.1.4-cuda9.0_0

```

```
packages:

alabaster==0.7.11
appdirs==1.4.3
asn1crypto==0.24.0
astroid==2.0.4
atomicwrites==1.2.1
attrs==18.2.0
Automat==0.7.0
Babel==2.6.0
backcall==0.1.0
bcolz==1.2.1
beautifulsoup4==4.6.3
bleach==2.1.4
bokeh==0.13.0
certifi==2018.8.24
cffi==1.11.5
chardet==3.0.4
click==6.7
click-plugins==1.0.3
cliff==2.8.2
cligj==0.4.0
cloudpickle==0.5.5
cmd2==0.9.4
colorama==0.3.9
configparser==3.5.0
constantly==15.1.0
cryptography==2.3.1
cryptography-vectors==2.3.1
cssselect==1.0.3
cycler==0.10.0
cymem==1.31.2
cytoolz==0.9.0.1
dask==0.19.0
decorator==4.3.0
descartes==1.1.0
dill==0.2.8.2
distributed==1.23.0
docutils==0.14
en-core-web-sm==2.0.0
entrypoints==0.2.3
feather-format==0.4.0
feedparser==5.2.1
Fiona==1.7.10
GDAL==2.2.2
geopandas==0.4.0
graphviz==0.9
h5py==2.8.0rc1
heapdict==1.0.0
html5lib==1.0.1
hyperlink==17.3.1
idna==2.7
imagesize==1.1.0
incremental==17.5.0
ipykernel==4.9.0
ipython==6.5.0
ipython-genutils==0.2.0
ipywidgets==7.4.1
isort==4.3.4
isoweek==1.3.3
jedi==0.12.1
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-contrib-core==0.3.3
jupyter-contrib-nbextensions==0.5.0
jupyter-core==4.4.0
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.4
jupyter-nbextensions-configurator==0.4.0
kaggle-cli==0.12.13
keyring==13.2.1
kiwisolver==1.0.1
lazy-object-proxy==1.3.1
locket==0.2.0
lxml==4.0.0
MarkupSafe==1.0
matplotlib==2.2.3
mccabe==0.6.1
MechanicalSoup==0.8.0
mistune==0.8.3
mizani==0.4.6
mkl-fft==1.0.6
mkl-random==1.0.1
more-itertools==4.3.0
msgpack==0.5.6
msgpack-numpy==0.4.3.1
munch==2.3.2
murmurhash==0.28.0
nbconvert==5.3.1
nbformat==4.4.0
notebook==5.6.0
numexpr==2.6.6
numpy==1.15.1
numpydoc==0.8.0
olefile==0.45.1
opencv-python==3.4.2.17
packaging==17.1
palettable==3.1.1
pandas==0.23.4
pandas-summary==0.0.5
pandocfilters==1.4.2
parso==0.3.1
partd==0.3.8
path.py==11.0.1
patsy==0.5.0
pbr==4.2.0
pexpect==4.6.0
pickleshare==0.7.4
Pillow==5.2.0
plac==0.9.6
plotnine==0.4.0
pluggy==0.7.1
preshed==1.0.0
prettytable==0.7.2
progressbar2==3.34.3
prometheus-client==0.3.0
prompt-toolkit==1.0.15
psutil==5.4.7
py==1.6.0
pyarrow==0.10.0
pyasn1==0.4.4
pyasn1-modules==0.2.1
pycodestyle==2.4.0
pycparser==2.18
pyflakes==2.0.0
Pygments==2.2.0
PyHamcrest==1.9.0
pylint==2.1.1
pyOpenSSL==18.0.0
pyparsing==2.2.0
pyperclip==1.6.4
pyproj==1.9.5.1
pyreadline==2.1
PySocks==1.6.8
pytest==3.7.4
python-dateutil==2.7.3
python-utils==2.3.0
pytz==2018.5
pywin32==223
pywinpty==0.5.4
PyYAML==3.13
pyzmq==17.1.2
QtAwesome==0.4.4
qtconsole==4.4.1
QtPy==1.5.0
regex==2017.11.9
requests==2.19.1
rope==0.11.0
scikit-learn==0.19.2
scipy==1.1.0
seaborn==0.9.0
Send2Trash==1.5.0
service-identity==17.0.0
Shapely==1.6.4.post2
simplegeneric==0.8.1
six==1.11.0
sklearn-pandas==1.7.0
snowballstemmer==1.2.1
sortedcontainers==2.0.4
spacy==2.0.12
Sphinx==1.7.8
sphinxcontrib-websupport==1.1.0
spyder==3.3.1
spyder-kernels==0.2.6
statsmodels==0.9.0
stevedore==1.29.0
tables==3.4.4
tblib==1.3.2
termcolor==1.1.0
terminado==0.8.1
testfixtures==6.3.0
testpath==0.3.1
thinc==6.10.3
toolz==0.9.0
torch==0.4.1
torchtext==0.2.3
torchvision==0.2.1
tornado==4.5.3
tqdm==4.24.0
traitlets==4.3.2
Twisted==18.7.0
typed-ast==1.1.0
ujson==1.35
urllib3==1.23
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==3.4.1
win-inet-pton==1.0.1
wincertstore==0.2
wrapt==1.10.11
zict==0.1.3
zope.interface==4.5.0
```

**Comment by geekan** [comment]
After debugging for around 8 hours, I found the error:
- While monitoring nvidia-smi, it seems like when power usage reach 200W, the system halted and reboot. 
- The power supply is buy from PinDuoDuo (PDD), it says it has 500W, but it may be only support ~200W.
- After changing this power supply, the issue solved.

**Comment by MSC19950601** [comment]
😀😀😀

**Comment by ifplusor** [comment]
so show~

**Comment by gongxun0928** [comment]
ahhhhh

**Comment by jackluo923** [comment]
@geekan Unless you post the power supply's rating label, its hard to say whether this is a problem with power supply or user error. No 550w power supply on the market for general use are rated to supply 550w on the 12v rail, power must be split amongst other sources: 3.3v, 5v, -12v..etc. Some cheap psu may even split the 12v into 2 power rails. Thus single 12v power rail with ~200w could just be the power supply's actual rating. I.e Nothing is wrong with the power supply and it halted due to exceeding power rating (which is rather good behavior).

**Comment by comzyh** [comment]
lol

**Comment by github410117** [comment]
hhh，拼夕夕

**Comment by hexinatgithub** [comment]
吃瓜群众，不过好好笑

**Comment by xuxiaoyuspallow** [comment]
hah, so it is a hardware problem.

**Comment by orca-zhang** [comment]
m

**Comment by qiangliqie** [comment]
so show....

**Comment by song940** [comment]
我大致翻译一下吧：

首先是有个同学在电脑上运行了一个叫做 fastai 的项目，而这个程序偶尔会导致系统死机(halt)。

于是这个同学去 GitHub issues 反馈了这个问题：

每当我训练模型时，系统就会死机重启。
whenever I tried to train a model(calling fit method), 
the system halted and then rebooted.

https://github.com/fastai/fastai/issues/751

同时，这位同学也提供了充分的材料表明这个问题绝对是 fastai 的 bug，
因为在运行 tensorflow 时就完全没有这个问题。

https://github.com/fastai/fastai/issues/751#issuecomment-419026267

几个小时后这位同学终于找到了问题原因：

￼

https://github.com/fastai/fastai/issues/751#issuecomment-419053678

翻译一下：

After debugging for around 8 hours, I found the root cause:
我鼓捣了 8 个多小时，终于找到原因了：

* TensorFlow and PyTorch's examples use only ~100W power.
* TensorFlow 和 PyTorch's 的例子大概消耗 ~100W 左右的电力.
* Fastai's example (lesson1) use 80~300W power (not stable, changing fast).
* Fastai's example (lesson1) 使用大概 80~300W 电力 (不稳定，变化比较大).
* While monitoring nvidia-smi, it seems like when power usage reach 200W, the system halted and rebooted.
* 直到我监控 nvidia-smi, 这个破玩意最多竟然可以干到 200W, 然后我的系统就挂了QAQ
* The power supply was bought from PinDuoDuo (Nasdaq: PDD), it says it support 550W, but it may only support ~200W, I doubt it's a fake one.
* 我现在使用的电源适配器是从「拼夕夕」买的 (Nasdaq: PDD), 他说能支持 550W, 但实际上最多只有 ~200W 左右, 这TM一定是假的吧。
* After changing the power supply, the issue has been solved.
* 反正我换一个就好了（手动微笑）。

**Comment by lovesmall** [comment]
@jackluo923 @geekan Just need post the PSU price.

**Comment by qianzui** [comment]
吃瓜群众路过

**Comment by deepbluez** [comment]
常逛github的码农按理说不该去拼多多买东西

**Comment by GaryChangCN** [comment]
Praise for the spirit of this research

**Comment by YL928** [comment]
拼夕夕受害者。

**Comment by hzh0292** [comment]
看到上了cnbeta特意过来留个言

**Comment by vitrum** [comment]
LOL🤣

**Comment by RuiZhang1993** [comment]
LOL

**Comment by feitxue** [comment]
意料之外,情理之中.

**Comment by liang37038** [comment]
笑死

**Comment by qazhuhuihao** [comment]
拼夕夕受害者。

**Comment by StillHanWind** [comment]
这个时候微笑就好

**Comment by huifrank** [comment]
原来拼多多不只是国内在用啊  hhhhh

**Comment by qdmfeng** [comment]
微博观光团

**Comment by opencrm** [comment]
脑子是个好东西，希望公司行政、财务能有，这也算是因小失大的正面教材😂很好奇，拼夕夕自己公司用不用自己的产品呢？😝

**Comment by suntiago** [comment]
你们公司在五环外吧！

**Comment by zgpg** [comment]
lol

**Comment by 123xiao** [comment]
PDD Niubi

**Comment by mingshen2018** [comment]
虽然我对拼多多没什么好感，但是这个太明显是黑文了，首先，电源供电不足导致的死机一般是不会重启的，更有甚者，会直接导致不能开机。其次，买东西的时候难道没看到电源的铭牌吗？自己不懂去怪商家，这黑的有点过分了吧，如果这个题主在京东或者亚马逊买个多路电源，同样导致这种问题，难道也说这个电源是假的吗？多路电源有多路的用途，自己买之前没有看好，然后怪商家说是假货，这就有点像是专门黑拼多多的了。

**Comment by orca-zhang** [comment]
楼上的小兄弟较真了啊，电源虚标是很正常的问题，频繁重启也是正常表现，和CPU过热一样。早年攒机盛行的年代，各大电脑城老板的羊毛出处就是电源和机箱，因为其他硬件价格基本都很透明，一般人都不太会关注这两个边缘配件，电源和机箱一般不会精确到型号甚至品牌，视你砍价程度来定最终型号和质量。

**Comment by ZYiDa** [comment]
并夕夕

**Comment by firstrose** [comment]
楼上说并夕夕国外也用的那位，你没看到楼主的资料写的是深圳么？

**Comment by opencrm** [comment]
一般专业人士，都知道电源有虚标部分，一般攒机电源功率会预留大于实际主机总功率的百分之2、30(大约值)，让电源持有一定富余的功率闲置在那，以保证硬件系统运行的稳定性。此外，电源也分不同等级的，有铜牌、金牌等等级认证等。从一般意义而言，档次越高的电源虚标部分会越小，相应的价格也越高，一般意义来说合格电源的大部分定价范围是1W>=1元，所以去拼夕夕买电源本身就是有问题的，连在JD这种大的电商买3C都会有小概率的遭遇二手东的经历，何况拼夕夕这种性质、特点分明的low bee电商，就不应该去，去拼夕夕本身就是忽略商品的稳定与品质的（用拼夕夕的理念说叫：价格敏感）。所以推测题主公司大概是那种小公司，或者碰到了在意金钱成本大于体验、时间成本的行政或财务（俗称pig队友），才会遭遇这种事吧。若是技术人员做的主机硬件配置选型，我只能说这技术不够专业，3C我首选JD（东哥，广告费记得接一下😝），碰到次品、二手东选择退换货就好，没这么麻烦，二手尤可用（也可换），JD自有物流售后还是不错的（早期至今对物流的钱不是白砸的），但假货在拼夕夕如何维权，如何鉴定，这连淘宝也没法杜绝的问题，在此就不予置评了。话说回来，来回折腾一来二去时间就这么过去了，毕竟这维权、退换货的时间本可用来创造更大商业价值。不吹不黑，Over😝

**Comment by zl939144892** [comment]
我来解释一下, 这款电源的名字就叫550W, 但是实际功率为200W, 哈哈哈哈, 编不下去了, 哈哈哈
Let me explain, the name of this power supply is called 550W, but the actual power is 200W, hahahaha, can't be edited, hahaha

**Comment by yucopowo** [comment]
赞楼看笑话

**Comment by lycbug666** [comment]
lol

**Comment by ejunjsh** [comment]
v2ex 过来，笑死了

**Comment by woainiming** [comment]
lol...

**Comment by chenzhangyi** [comment]
你好， 我是拼多多这边的开发， 方便的话麻烦提供下你所购买电源的链接或者订单号，我找同事看一下这个商品是不是有问题.

**Comment by ObserverAlpha** [comment]
楼上有大佬来了😂

**Comment by fsword73** [comment]
其实这个问题可能不是550watt电源的问题。 
NVIDIA显卡的峰值功耗是平均功耗的1.5甚至3倍。如果nvidia-smi报告了300watt,那么它的峰值功耗可能是450watts,考虑到系统其它部分功耗，CPU,  Memory, motherboard,  KeyBoards, Fan and etc.  那么需要的峰值功耗有可能超过600瓦。如果峰值功耗是三倍的平均功耗，那就更惨了，实际上需要1000watts的电源。

**Comment by fsword73** [comment]
@geekan .  Translate into english:
   The issue could not be the 550 watts' power supplies. 
    GPU's peak power could be 1.5x to 3x of TDP or average.  if nvidia-smi reports 300 watts, the peak power could be 450wats with 1.5x.  plus the other components' power , CPU, motherboard, disk, and etc, the peak power of whole system could exceeds 600 watts.  if the peak power is 3x of TDP,   the whole system even needs 1000 watts . 

other brands' power supplies may not solve the issue.  The good solution is to buy 800 watts to 1000 watts's power supply.  If you want  to use 2 GPU cards for AI trainning, 1500 watts are better. 4 GPUs,  3000 watts are suggested

**Comment by qinlili23333** [comment]
前来围观骈夕夕

**Comment by yinghualuowu** [comment]
PDD? NO! This is  PXX >W<

**Comment by skytsui** [comment]
夕夕夕

**Comment by xiaofuzoe** [comment]
前来围观拼夕夕 PDD

**Comment by ArtemisSaber** [comment]
Do not ever buy anything of importance from pdd
千万不要在拼多多买任何重要一点的东西

**Comment by panda8z** [comment]
拼夕夕惨案

**Comment by Heiyuan** [comment]
观光.

**Comment by Z-fly** [comment]
观光.

**Comment by jasnzhuang** [comment]
happy ending~~~ahhhh
plz do not trust PDD

**Comment by ijiroterunao** [comment]
观光

**Comment by Dead-Rabbit** [comment]
路过

**Comment by 70852** [comment]
终于知道 并夕夕 这个梗的意思了。

**Comment by 70852** [comment]
并夕夕

**Comment by Akimoto-Cris** [comment]
笑死

**Comment by qixiaobo** [comment]
> * PDD

Interesting
